tools/spack-scripting/scripting/cmd/installcosmo.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+import os
+import subprocess
+import shutil
+
+import llnl.util.tty as tty
+import re
+
+# same used by spack, comes with it
+import ruamel.yaml as yaml
+
+import spack.config
+import spack.cmd
+import spack.cmd.common.arguments as arguments
+import spack.repo
+from spack.stage import DIYStage
+from spack.spec import Spec
+from spack.cmd.dev_build import dev_build
+from spack.main import SpackCommand
+
+description = "Install cosmo and dycore with or without testing using spec.yaml for key dependencies"
+section = "scripting"
+level = "long"
+
+
+def setup_parser(subparser):
+    arguments.add_common_arguments(subparser, ["jobs"])
+
+    subparser.add_argument(
+        "-t", "--test", action="store_true", help="Dev-build with testing"
+    )
+    arguments.add_common_arguments(subparser, ["spec"])
+
+
+def custom_devbuild(spec, jobs):
+    package = spack.repo.get(spec)
+
+    if package.installed:
+        package.do_uninstall(force=True)
+
+    package.do_install(verbose=True, make_jobs=jobs)
+
+
+def installcosmo(self, args):
+    # Extract and concretize cosmo_spec
+    if not args.spec:
+        tty.die("spack dev-build requires a package spec argument.")
+
+    specs = spack.cmd.parse_specs(args.spec)
+    if len(specs) > 1:
+        tty.die("spack dev-build only takes one spec.")
+
+    cosmo_spec = specs[0]
+
+    # Collect user-specified versions of dependencies before first concretization
+    user_versioned_deps = set()
+    for dep in cosmo_spec.traverse():
+        if dep.name != "cosmo" and dep.versions and len(dep.versions) != 0:
```

Comment:
I was just thinking... that the requirements and implementation of the installcosmo is not really specific to cosmo, and that we would like to probably solve the same problem for any other package, e.g. int2lm ? 
If so we could rename the command again (in order to be able to use for any package) ? 
Only this line has a hardcoded "cosmo". anyway to make it general ?
@Stagno @jonasjucker

Reply:
I see it as very specific for package `cosmo`: we are setting the version of `cosmo-dycore` to the same as `cosmo`'s and we are looking for `spec.yaml` in `/cosmo/ACC/spack/`. 
I don't think it can be generalized to any package, due to these very specific requirements.

Reply:
I agree with @Stagno as well.

Reply:
ok, but these are easy. I thought we could pass the yaml file as an argument with -f cosmo/ACC/spec.yaml (at least I remember that sometime in the proposal f Giacomo)
Also for cosmo-dycore we could add the logic to add the dependency only if the package == 'cosmo'. 
The problem is more pressing for cosmo, but not specific for cosmo. Same will arise when we want to control the versions of the dependencies of int2lm or any other package. 
Will we create install<pkg> for every package we want to solve this way?

Reply:
But I think we can discuss at the next meeting, and see if we change anything with the different PR. 
I think it is fine to merge this already like this. 
I will merge as soon as @Stagno take a look at the two suggestions below

Reply:
Let's discuss tomorrow yes. I accepted the suggestions, good to go for me.